Aspire Budgeting/ViewModels/FileSelectorViewModel.swift

```diff
@@ -0,0 +1,51 @@
+//
+// FileSelectorViewModel.swift
+// Aspire Budgeting
+//
+
+import Foundation
+
+
+struct FileSelectorViewModel {
+  enum State {
+    case isLoading
+    case filesRetrieved
+    case error
+  }
+
+  let fileManagerState: RemoteFileManagerState
+
+  typealias FileSelectedCallBack = (File) -> Void
+  let fileSelectedCallback: FileSelectedCallBack?
+
+  var currentState: State {
+    switch fileManagerState {
+    case .isLoading:
+      return .isLoading
+
+    case .filesRetrieved(_):
+      return .filesRetrieved
+
+    case .error(_):
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/aspirebudgetingmobile/aspirebudgeting_ios/blob/ff44934e95fcfbf9892d0e49040b4320aa9b4509/Aspire Budgeting/ViewModels/FileSelectorViewModel.swift#L29"></span>Arguments can be omitted when matching enums with associated types if they are not used.<br />
<code>empty_enum_arguments</code> <code>FileSelectorViewModel.swift:29</code></td>
    </tr>
  </tbody>
</table>